docs/DeveloperGuide.md

```diff
@@ -521,330 +523,322 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+    * 1b1. SudoHR shows an error message.
 
+      Use case resumes at step 1.
 
-**Use case: UC12 - Update event**
-**MSS:**
-1. User requests to __list events(UC6)__
-2. SudoHR shows all events and their details
-3. User requests to update event details on SudoHR
-4. SudoHR update event.
-5. Use case ends.
+* 1c. The employee has already taken leave on any of the dates in the range of the dates.
+    * 1c1. SudoHR shows an error message.
 
-**Extensions**
+      Use case resumes at step 1.
 
-* 2a. The list is empty.
+* 1d. The end date is more than 6 days away from the start date.
+    
+    * 1d1. SudoHR shows an error message.
 
-  Use case ends.
+      Use case resumes at step 1.
 
-* 3a. The given index is invalid.
+* 1e. There is no employee with the eid given as input.
+   
+    * 1e1. SudoHR shows an error message.
 
-    * 3a1. SudoHR shows an error message.
+      Use case resumes at step 1.
 
-      Use case resumes at step 2.
+* 1f. The end date is before start date.
+    
+    * 1f1. SudoHR shows an error message.
 
-* 3b. The given argument is empty.
+      Use case resumes at step 1.
+    
 
-    * 3b1. SudoHR shows an error message.
+**Use case: UC8 - Delete employee's leave**
+**MSS:**
+1. User __lists all leaves taken by employee(UC 11).__
+2. User requests to delete leave from SudoHR for the employee on a specific day.
+3. SudoHR informs user that employee's leave has been successfully deleted.
 
-      Use case resumes at step 2.
+    Use case ends.
 
+**Extensions**
 
-**Use case: UC13 - List event**
-**MSS:**
-1. User requests SudoHR to list all events
-2. SudoHR shows all events and their details
+* 1a. The list is empty.
 
+      Use case ends.
 
-**Use case: UC14 - Add employee to event**
-**MSS:**
-1. User requests to __list events(UC6)__
-2. SudoHR shows all events and their details
-3. User requests to list employees
-4. SudoHR shows all employees and their details
-5. User requests to add employee to event on SudoHR
-6. SudoHR adds employee to event.
-7. Use case ends.
+* 2a. The given command has invalid inputs.
 
-**Extensions**
+    * 2a1. SudoHR shows an error message.
 
-* 2a. The event list is empty.
+      Use case resumes at step 3.
 
-  Use case ends.
+* 2b. The given command has missing inputs.
+    * 2b1. SudoHR shows an error message.
 
-* 4a. The employee list is empty.
+      Use case resumes at step 3.
 
-  Use case ends.
+* 2c. The employee has not taken leave on the specific day
+    * 2c1. SudoHR shows an error message.
+
+      Use case resumes at step 3
 
-* 5a. The given Employeeindex is invalid.
+* 2d. There is no employee with the eid given as input.
+    * 2d1. SudoHR shows an error message.
 
-    * 5a1. SudoHR shows an error message.
+      Use case resumes at step 3 
 
-      Use case resumes at step 4.
 
-* 5b. The given Eventindex is invalid.
+**Use case: UC9 - List leaves**
+**MSS:**
+1. User requests SudoHR to list all days where any employee has taken leave
```

Comment:
Since we persist the leaves in which there are no employees, this statement would not be accurate i think, maybe can consider adding another statement: or has been considered for leave before

Reply:
Statement is accurate since predicate used filters out empty leaves

Reply:
ma bad